returnn/datasets/distrib_files.py

```diff
@@ -324,9 +306,7 @@ def _get_sub_dataset_dict(self, files: List[FileTree]) -> Tuple[Dict[str, Any],
                 f"{self}: sub dataset should have explicit seq_ordering "
                 f"(or seq_order_control_dataset for MetaDataset), got: {dataset_dict}"
             )
-        self._lazy_init_file_cache_proc()
-        dataset_dict, exit_hook = self._file_cache.handle_cached_files_in_config(dataset_dict)
```

Comment:
This is where the existing code would block the master process.

Reply:
For reference, I made #1549 to keep track of this issue.

But I think we can resolve this thread here.